crates/cubecl-linalg/src/matmul/components/tile/plane.rs

```diff
@@ -165,22 +163,40 @@ impl<I: Numeric, O: Numeric, T: tile::Config, const M: u32, const N: u32, const
 
         let offset = row_offset + unit % row_division * num_lines;
 
-        if comptime!(line_size == 1) {
-            for col in 0..num_lines {
-                slice[col + offset] = Line::cast_from(out[col]);
-            }
-            // TODO: very obscure bug, fails on wgpu if we don't do the loop twice
-            for col in 0..num_lines {
-                slice[col + offset] = Line::cast_from(out[col]);
-            }
-        } else {
+        if comptime!(line_size >= 4) {
+            #[unroll]
             for col in 0..num_lines {
                 let mut line = Line::<C>::empty(line_size);
+                #[unroll]
                 for j in 0..line_size {
                     line[j] = C::cast_from(out[col * line_size + j]);
                 }
                 slice[col + offset] = line;
             }
+        } else {
+            // There are weird behaviours on wgpu on metal with vec1 and vec2,
+            // where some values are left empty.
+            // This is patched by repeating loops or deactivating unrolling
+
+            if comptime!(line_size == 1) {
+                #[unroll]
+                for col in 0..num_lines {
+                    slice[col + offset] = Line::cast_from(out[col]);
+                }
+                // It seems we must repeat this loop without unroll
+                for col in 0..num_lines {
+                    slice[col + offset] = Line::cast_from(out[col]);
+                }
```

Comment:
Can we just remove the unroll instead?

Reply:
Sadly no, I know it makes no sense 😢 This is the bug that appears:
`---- tests::plane_mma_matmul::f32_ty::test_stage_matmul_s8x1x1 stdout ----
thread 'tests::plane_mma_matmul::f32_ty::test_stage_matmul_s8x1x1' panicked at /Users/louisfortier-dubois/Projects/cubecl/crates/cubecl-linalg/src/matmul/tests/cmma_matmul/matmul_test_launcher.rs:204:9:
Values differ more than epsilon: index=38 actual=-0.000000000000000007134327, expected=1.3014265, difference=1.3014265, epsilon=0.0001
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::plane_mma_matmul::flex32_ty::test_stage_matmul_s8x1x1 stdout ----
thread 'tests::plane_mma_matmul::flex32_ty::test_stage_matmul_s8x1x1' panicked at /Users/louisfortier-dubois/Projects/cubecl/crates/cubecl-linalg/src/matmul/tests/cmma_matmul/matmul_test_launcher.rs:204:9:
Values differ more than epsilon: index=38 actual=-25291050000000, expected=1.3014265, difference=25291050000000, epsilon=0.8192`

Really don't know what's special with index 38 here.